flocker/provision/_gce.py

```diff
@@ -0,0 +1,494 @@
+        calls to the GCE API.
+    """
+    address = field(type=bytes)
+    private_address = field(type=bytes)
+    distribution = field(type=bytes)
+    project = field(type=unicode)
+    zone = field(type=unicode)
+    name = field(type=unicode)
+    compute = field()
+
+    def get_default_username(self):
+        return bytes(_GCE_ACCEPTANCE_USERNAME)
+
+    def provision(self, package_source, variants):
+        commands = []
+        sed_str = (
+            r"s/^\(PermitRootLogin *no\)/"
+            r"\n"
+            r"# Commented out by flocker provisioner:\n"
+            r"#\1\n"
+            r"/"
+        )
+
+        # On GCE, the centos image is configured to not allow root to ssh on.
+        # Enable the configuration and restart the daemon on that system.
+        if self.distribution.startswith("centos"):
+            username = self.get_default_username()
+            enable_root_ssh = sequence([
+                sudo_from_args(
+                    ['sed', sed_str, '-i.backup', '/etc/ssh/sshd_config']
+                ),
+                sudo_from_args(['systemctl', 'restart', 'sshd']),
+            ])
+            commands.append(retry_effect_with_timeout(
+                run_remotely(
+                    username=username,
+                    address=self.address,
+                    commands=enable_root_ssh,
+                ),
+                30
+            ))
+
+        commands.append(provision_for_any_user(self, package_source, variants))
+        return sequence(commands)
+
+    def destroy(self):
+        with start_action(
+            action_type=u"flocker:provision:gce:destroy",
+            instance_id=self.name,
+        ):
+            operation = self.compute.instances().delete(
+                project=self.project,
+                zone=self.zone,
+                instance=self.name
+            ).execute()
+            wait_for_operation(self.compute, operation, timeout=60)
+
+    def reboot(self):
+        """
+        I think this is never called, and can probably be removed. If it were
+        to be implemented it would look something like the following:
+
+            operation = self.compute.instances().reset(
+                project=self.project,
+                zone=self.zone,
+                instance=self.name
+            ).execute()
+            wait_for_operation(self.compute, operation, timeout=60)
+
+        But, as that is untested, it will remain unimplemented.
+        """
+        raise NotImplementedError(
+            "GCE does not have reboot implemented because it has "
+            "experimentally been determined to not be needed.")
+
+
+@implementer(IProvisioner)
+class GCEProvisioner(PClass):
+    """
+    A provisioner that can create instances on GCE.
+
+    :ivar unicode zone: The zone in which instances will be provisioned in.
+    :ivar unicode project: The project under which instances will be
+        provisioned in.
```

Comment:
It appears as if I'm trying a little too hard to end my sentences in a preposition in.